src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -19,7 +19,17 @@ public MessageConverter(Func<BasicDeliverEventArgs, string> messageIdStrategy)
             this.messageIdStrategy = messageIdStrategy;
         }
 
-        public string RetrieveMessageId(BasicDeliverEventArgs message) => messageIdStrategy(message);
+        public string RetrieveMessageId(BasicDeliverEventArgs message, Dictionary<string, string> headers)
+        {
+            var messageId = messageIdStrategy(message);
+
+            if (string.IsNullOrWhiteSpace(messageId) && !headers.TryGetValue(Headers.MessageId, out messageId))
+            {
+                throw new InvalidOperationException("The message id strategy did not provide a message id, and the message does not have an 'NServiceBus.MessageId' header.");
```

Comment:
The wording here should probably "ID" here instead of "id".

Though that means we have the same problem in the xml comments of `CustomMessageIdStrategy` as well. Maybe go ahead and clean up both now?